src/core/execute.c

```diff
@@ -846,6 +846,12 @@ static int setup_pam(
                         goto fail;
         }
 
+        STRV_FOREACH(e, *env) {
+                pam_code = pam_putenv(handle, *e);
+                if (pam_code != PAM_SUCCESS)
+                        goto fail;
```

Comment:
> Edit: on closer inspection, our e is NULL-terminated (it is a pointer into a vector returned by strv_env_merge) and should not be able to cause this bug.

Well, we `strv_free(accum_env)` twice:
* https://github.com/joukewitteveen/systemd/blob/daf3aaeebfc991a337af0372206257dc204b25de/src/core/execute.c#L994
* https://github.com/joukewitteveen/systemd/blob/daf3aaeebfc991a337af0372206257dc204b25de/src/core/execute.c#L1787 (note `_cleanup_strv_free_` [there](https://github.com/joukewitteveen/systemd/blob/daf3aaeebfc991a337af0372206257dc204b25de/src/core/execute.c#L1473))


> #3680 

Thanks! I'll take a look.